src/unity/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -126,19 +163,39 @@ def create(dataset, target, feature, max_iterations=10, verbose=True, batch_size
                     loss = softmax_cross_entropy_loss(z, y)
                     # Backpropagate the error for one iteration.
                     loss.backward()
-                    outputs.append(z)
-            # Updates internal evaluation
-            metric.update(label, outputs)
             # Make one step of parameter update. Trainer needs to know the
             # batch size of data to normalize the gradient by 1/batch_size.
             trainer.step(batch.data[0].shape[0])
+        train_data.reset()
 
-        _, accuracy = metric.get()
+        # Calculate training metric
+        for batch in train_data:
+            data = _mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
+            label = _mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
+            outputs = [custom_NN(x) for x in data]
+            train_metric.update(label, outputs)
+        train_data.reset()
+
+        # Calculate validataion metric
+        for batch in validation_data:
+            data = _mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
+            label = _mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
+            outputs = [custom_NN(x) for x in data]
+            validation_metric.update(label, outputs)
```

Comment:
This for loop and the one above could probably be two function calls.

Reply:
It certainly could be, but I don't think it's worth it. I guess it would look something like this:

```python
def compute_accuracy(data, metric):
    for batch in data:
        features = _mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
        y = _mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
        outputs = [custom_NN(x) for x in features]
        metric.update(label, outputs)
```

```python
compute_accuracy(train_data, train_metric)
train_data.reset()
compute_accuracy(validation_data, validation_metric)
```
It's only a savings of two lines. Not really sure that's worth an inner function that isn't completely straight forward. What do you think?

Reply:
Absolutely, I wouldn’t hold up merging this PR for that change

Reply:
I agree that it's more readable as is.